aptos-move/framework/aptos-stdlib/sources/data_structures/ring_deque.move

```diff
@@ -0,0 +1,291 @@
+/// Ring buffer implementation of a double-ended queue.
+///
+/// Has O(1) push/pop operations at both front and back, unlike a vector
+/// which has O(n) push/pop operations at the front.
+module aptos_std::ring_deque {
+
+    use std::option::{Self, Option};
+    use std::vector;
+
+    /// Buffer capacity must be nonzero.
+    const E_CAPACITY_ZERO: u64 = 0;
+    /// Ring deque is full;
+    const E_FULL: u64 = 1;
+    /// Ring deque is empty;
+    const E_EMPTY: u64 = 2;
+
+    struct RingDeque<T> has copy, drop, store {
```

Comment:
We're thinking of converting most core data structures to native implementations in Rust for better performance. I think this might also benefit from a similar treatment

Reply:
Would that conversion from non native to native require action from the dev, if would it be totally transparent? If the latter, is there any risk in adding this non native implementation now?